Yosemite/Yosemite/PointOfSale/PointOfSaleCouponService.swift

```diff
@@ -73,7 +76,21 @@ public final class PointOfSaleCouponService: PointOfSaleItemServiceProtocol {
 
     @MainActor
     public func providePointOfSaleItems(pageNumber: Int) async throws -> PagedItems<POSItem> {
-        let coupons = providePointOfSaleCoupons()
+        let coupons = await providePointOfSaleCoupons()
```

Comment:
Idea: If coupons is empty, we could wait for `syncCouponsFromRemote` to complete before returning `PagedItems`.

Reply:
That sounds good, updated here: https://github.com/woocommerce/woocommerce-ios/pull/15423/commits/1a23559030cd488ee59b5bb26ade35d15fab4290 . I was a bit wary of using a detached Task attached to a dispatcher action, but from testing seems to work fine, let me know if raises any red flag:

- If non empty, we return storage and fire a sync task which will complete at some point
- If empty, we wait for the sync both remote and storage, then return that